CapsuleKey should eagerly impl Hash, Eq, PartialEq, and Debug

I thought this was a one-line fix with `#derive` but apparently not. I'm running into an issue with this now:

```
error[E0119]: conflicting implementations of trait `From<CapsuleKey>` for type `CapsuleKey`
  --> rearch/src/capsule_key.rs:19:1
   |
19 | impl<T: Hash + Eq + Debug + Send + Sync + 'static> From<T> for CapsuleKey {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T> From<T> for T;
```

That's a little annoying. I can remove any one of the derived traits and then it compiles fine, but that really feels like a hack. I want to keep that From impl since it is really handy, so I am wondering if I should just drop the `Hash` impl so it all still compiles fine.

--- a/rearch/src/capsule_key.rs
+++ b/rearch/src/capsule_key.rs
@@ -14,7 +14,7 @@ use crate::Capsule;
 /// Most applications are just fine with static (function) capsules.
 /// If you are making an incremental computation focused application,
 /// then you may need dynamic capsules and the [`From`] impl.
-#[derive(Default)]
+#[derive(Debug, Default, PartialEq, Eq)]
 pub struct CapsuleKey(CapsuleKeyType);
 impl<T: Hash + Eq + Debug + Send + Sync + 'static> From<T> for CapsuleKey {
     fn from(key: T) -> Self {

